src/conventional/ionosonde_ascii2ioda.py

```diff
@@ -0,0 +1,415 @@
+            else:
+                logging.error("Unknown model should be one of: [ART, POL]")
+                break
+            local_data['Slope'] = np.append(local_data['Slope'], Slope)
+            local_data['polanLayer'] = np.append(local_data['polanLayer'], POLANL)  # removed from final IODA file
+            # populate the first row of ObsValues
+            local_data = populate_obsValue(line, local_data, model=model)
+
+            header_read = True
+        except StopIteration:
+            # If StopIteration is raised, break from the loop
+            break
+
+    return local_data, header_read
+
+
+def populate_obsValue(line, local_data, model='ART'):
+    # get the electron density as a funtion of height/range and chirp frequency
+    # if can correctly parse all fields populate local_data else do nothing
+    # ObsValue data row (example)
+    # height    freq  f_conf      density density_conf  ARTIST   POLAN
+    # 186.00  -99.00  -99.00  -0.9900E+02  -0.9900E+02  0.00 c   2.04 F
+
+    try:
+        height, freq, f_conf, density, density_conf, ARTScale, Layer, POLAN, polanLayer = line.split()
+
+        local_data['height'] = np.append(local_data['height'], height)
+        local_data['frequency'] = np.append(local_data['frequency'], freq)
+        local_data['frequencyConfidence'] = np.append(local_data['frequencyConfidence'], f_conf)
+        local_data['electronDensity'] = np.append(local_data['electronDensity'], density)
+        local_data['electronDensityConfidence'] = np.append(local_data['electronDensityConfidence'], density_conf)
+        local_data['ionosondeScalingParameter'] = np.append(local_data['ionosondeScalingParameter'], ARTScale)
+        # the ionospheric layer/region is dependent on the model selected
+        if 'POL' in model:
+            ionosphericLayer = get_layer(polanLayer, polanLayer=local_data['polanLayer'][0], model=model)
+        else:
+            ionosphericLayer = get_layer(Layer, model=model)
+        local_data['ionosphericLayer'].append(ionosphericLayer)
+    except ValueError:
+        pass
+
+    return local_data
+
+
+def get_layer(Layer, polanLayer=1, model='ART'):
+    # this routine should define a consistent naming for the ionospheric region
+    # a conversation is needed with an ionosonde expert to define this
+    # a table representation of the layers may be beneficial to avoid storing strings
+
+    # depending on use of 'ART' or 'POL' set ionospheric layer/region
+    ionosphericLayer = int_missing_value
+    match model:
+        case 'ART':
+            # do not know what T means
+            if 'F1' in Layer:
+                ionosphericLayer = 1
+            elif 'F2' in Layer:
+                ionosphericLayer = 2
+            elif 'E' in Layer:
+                ionosphericLayer = 0
+            elif 'T' in Layer:
+                ionosphericLayer = 3
+            elif 'V' in Layer:
+                ionosphericLayer = 4
```

Comment:
T = topside of F2 layer, V = valley region (between E and F1 (or between E and F2)).
There is also 'c' = connecting region in the bottom of F2